okhttp/src/main/java/okhttp3/internal/connection/RealConnection.java

```diff
@@ -276,6 +280,9 @@ private void connectTls(ConnectionSpecSelector connectionSpecSelector) throws IO
             + "\n    subjectAltNames: " + OkHostnameVerifier.allSubjectAltNames(cert));
       }
 
+      X509Certificate cert = (X509Certificate) unverifiedHandshake.peerCertificates().get(0);
+      supportedHosts = new HashSet<>(OkHostnameVerifier.allSubjectAltNames(cert));
```

Comment:
nit: https://publicobject.com/2016/02/08/linkedhashmap-is-always-better-than-hashmap/